src/service/core/config/tests/test_configmap_loader_unit.py

```
@@ -343,6 +343,90 @@ def test_empty_configs_valid(self):
+
+    Background: Pydantic's default str(error) includes `input_value=<repr>`
+    for every rejected field. If a credential (e.g. access_key) is
+    rejected by a discriminator mismatch, the plaintext ends up in logs
+    and K8s Events. The loader's error formatter must redact those
+    values while preserving visibility for non-sensitive fields.
+    """
+
+    def _make_error(self, *field_errors) -> configmap_loader.pydantic.ValidationError:
+        """Build a ValidationError by defining a small model + running it."""
+        class FakeModel(configmap_loader.pydantic.BaseModel):
+            model_config = {'extra': 'forbid'}
```

Ruff (0.15.10)

[warning] Mutable default value for class attribute

(RUF012)